app/ptokens/models.py

```diff
@@ -120,4 +129,47 @@ def save(self, *args, **kwargs):
 
     @property
     def title(self):
-        return self.title
\ No newline at end of file
+        return self.title
+
+
+class RedemptionToken(SuperModel):
+    """Define the structure of a Redemption PToken"""
+
+    REDEMPTION_STATUS_CHOICES = [
+        ('request', 'requested'),
+        ('accepted', 'accepted'),
+        ('denied', 'denied'),
+        ('completed', 'completed')
+    ]
+    ptoken = models.ForeignKey(PersonalToken, null=True, on_delete=models.SET_NULL)
+    redemption_state = models.CharField(max_length=50, choices=REDEMPTION_STATUS_CHOICES, default='request', db_index=True)
+    network = models.CharField(max_length=255, default='')
+    reason = models.CharField(max_length=1000, default='')
+    total = models.DecimalField(default=0, decimal_places=2, max_digits=50, blank=True, null=True, help_text='Total ptokens to redeem')
+    txid = models.CharField(max_length=255, default='')
+    redemption_accepted = models.DateTimeField(null=True)
+    redemption_requester = models.ForeignKey(
+        'dashboard.Profile', null=True, on_delete=models.SET_NULL, related_name='redemptions', blank=True
+    )
+    tx_status = models.CharField(max_length=9, choices=TX_STATUS_CHOICES, default='na', db_index=True)
+    web3_created = models.DateTimeField(null=True)
+
+
+@receiver(post_save, sender=PersonalToken, dispatch_uid="PTokenActivity")
+def psave_ptoken(sender, instance, **kwargs):
+    pass
+
+
+class PurchasePToken(SuperModel):
+    ptoken = models.ForeignKey(PersonalToken, null=True, on_delete=models.SET_NULL)
```

Comment:
can you purchase a ptoken without any ptoken? the better here are not to use on_deleted=models.RESTRICTED to avoid loss information when occur accidental deletions?